tfmigrate/state_xmv_action.go

```diff
@@ -0,0 +1,129 @@
+package tfmigrate
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/minamijoyo/tfmigrate/tfexec"
+)
+
+// StateXMvAction implements the StateAction interface.
+// StateXMvAction moves a resource from source address to destination address in
+// the same tfstate file.
+type StateXMvAction struct {
+	// source is a address of resource or module to be moved which can contain wildcards.
+	source string
+	// destination is a new address of resource or module to move which can contain placeholders.
+	destination string
+}
+
+var _ StateAction = (*StateXMvAction)(nil)
+
+// NewStateMvAction returns a new StateXMvAction instance.
+func NewStateXMvAction(source string, destination string) *StateXMvAction {
+	return &StateXMvAction{
+		source:      source,
+		destination: destination,
+	}
+}
+
+// StateUpdate updates a given state and returns a new state.
+// Source resources have wildcards wich should be matched against the tf state. Each occurrence will generate
+// a move command.
+func (a *StateXMvAction) StateUpdate(ctx context.Context, tf tfexec.TerraformCLI, state *tfexec.State) (*tfexec.State, error) {
+	stateMvActions, err := a.generateMvActions(ctx, tf, state)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, action := range stateMvActions {
+		state, err = action.StateUpdate(ctx, tf, state)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return state, err
+}
+
+func (a *StateXMvAction) generateMvActions(ctx context.Context, tf tfexec.TerraformCLI, state *tfexec.State) (response []*StateMvAction, err error) {
+	stateList, err := tf.StateList(ctx, state, nil)
+	if err != nil {
+		return nil, err
+	}
+	return a.getStateMvActionsForStateList(stateList)
+}
+
+// When a wildcardChar is used in a path it should only match a single part of the path
+// It can therefore not contain a dot(.), whitespace nor square brackets
+const matchWildcardRegex = "([^\\]\\[\t\n\v\f\r ]*)"
```

Comment:
Could you explain the reason why you intentionally restricted the match pattern?
Strictly speaking, it is possible to include spaces in the resource address. An example can be found at: https://github.com/minamijoyo/tfmigrate/issues/5
I also suspect that a possible next feature request would be likely "allow selecting all with `*`".
Can we interpret the glob style wildcard `*` as a regular expression `.*`? If not, why?

Reply:
So, the reasoning was probably overthinking on my part. I narrowed my train of thoughts for the use case of moving resources that are rendered as part of a for each and also to allow moving to and from a for each pattern. In the moves I played in my head I only saw a use case where a part of the part gets replaced. For example having 2 named resources and going to a for each then the last part just changes.  When you have resources generated in a for each and want to introduce a module to make your code follow the 'DRY' -principle more then you also match the last part but if you want to refactor modules and move resources between them then you might need the part between the `["`  and `"]`.

But your suggestion indeed works equally well for those cases and is more generic. So will add it as well.